src/algorithm/crba.hxx

```diff
@@ -161,27 +161,41 @@ namespace se3
                      const se3::Model & model,
                      se3::Data & data)
     {
-      typedef typename SizeDepType<JointModel::NV>::template ColsReturn<Matrix6x>::Type ColsBlock;
+      typedef typename SizeDepType<JointModel::NV>::template ColsReturn<Data::Matrix6x>::Type ColsBlock;
       
       const Model::JointIndex & i = (Model::JointIndex) jmodel.id();
       const Model::Index & parent = model.parents[i];
       
       data.Ycrb[parent] += data.liMi[i].act(data.Ycrb[i]);
       
       jdata.U() = data.Ycrb[i] * jdata.S();
-      if (JointModel::NV == -1)
-      {
-        ColsBlock jF
-        = data.Ag.middleCols(jmodel.idx_v(), jmodel.nv());
-      }
-      else
-      {
-        ColsBlock jF
+
+      ColsBlock jF
         = data.Ag.middleCols <JointModel::NV> (jmodel.idx_v());
-      }
+
       forceSet::se3Action(data.oMi[i],jdata.U(),jF);
     }
     
+    static void algo(const se3::JointModelBase<JointModelComposite> & jmodel,
+                     se3::JointDataBase<JointDataComposite> & jdata,
+                     const se3::Model & model,
+                     se3::Data & data)
+    {
+      typedef SizeDepType<JointModel::NV>::ColsReturn<Data::Matrix6x>::Type ColsBlock;
+      
+      const Model::JointIndex & i = (Model::JointIndex) jmodel.id();
+      const Model::Index & parent = model.parents[i];
+      
+      data.Ycrb[parent] += data.liMi[i].act(data.Ycrb[i]);
+      
+      jdata.U() = data.Ycrb[i] * jdata.S();
+      
+      ColsBlock jF
```

Comment:
However, why is it necessary to reintroduce it here for Composite. It is a very bad design that algorithm should be specific to some joints. Dont you have a NV in joint composite? You should, it is possible to have a constant (known a compil time) from both included joints.

Please remove this specific code.